integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/DataSourceV2ScanRelationOnEndInputDatasetBuilder.java

```diff
@@ -50,10 +62,51 @@ public List<InputDataset> apply(DataSourceV2ScanRelation plan) {
     DataSourceV2Relation relation = plan.relation();
     DatasetCompositeFacetsBuilder datasetFacetsBuilder =
         new DatasetCompositeFacetsBuilder(context.getOpenLineage());
+
+    getInputStatisticsInputDatasetFacet(plan)
+        .ifPresent(stats -> datasetFacetsBuilder.getInputFacets().inputStatistics(stats));
+
     return DataSourceV2RelationDatasetExtractor.extract(
         factory, context, relation, datasetFacetsBuilder);
   }
 
+  private Optional<InputStatisticsInputDatasetFacet> getInputStatisticsInputDatasetFacet(
+      DataSourceV2ScanRelation plan) {
+    if (plan.scan() == null) {
+      return Optional.empty();
+    }
+    try {
+      List<ScanTask> tasks = (List<ScanTask>) MethodUtils.invokeMethod(plan.scan(), true, "tasks");
```

Comment:
Where is this coming from? I don't see it implemented for Scan interface: https://github.com/apache/spark/blob/381f3691bd481abc8f621ca3f282e06db32bea31/sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/Scan.java#L46

Reply:
If it's Iceberg, shouldn't it be separated somehow? Or, not at least, have some isolated code path.

Reply:
For Iceberg, I am planning a separate PR -> https://github.com/OpenLineage/OpenLineage/pull/3256/files which will introduce a separate facet. This PR is more generic but gets limited amount of metrics. For iceberg we can obtain much more information, but I think it's worth putting it into a separate iceberg specific facet. In some cases, we get both facets and I don't think this duplication is a problem at all.

Reply:
My point is about separating the specific Iceberg implementation of `getInputStatisticsInputDatasetFacet` in `DataSourceV2ScanRelation`, as it implements some Iceberg specific functionality. Maybe to some `IcebergUtils` if we don't have `vendor` code for it yet.

Reply:
Ohh,  you're right, now I get it. I will remove feature for iceberg and bring it back in the separate PR.